qiskit_ibm_runtime/base_primitive.py

```diff
@@ -194,7 +185,17 @@ def session(self) -> Optional[Session]:
         Returns:
             Session used by this primitive, or ``None`` if session is not used.
         """
-        return self._session
+        deprecate_function("session", "0.23.0", "Please use the 'mode' property instead.")
+        return self._mode
+
+    @property
+    def mode(self) -> Optional[Session | Batch | str]:
```

Comment:
I don't think `self._mode` can be a `str`. 

```suggestion
    def mode(self) -> Optional[Session | Batch]:
```